src/Classes/DBPopulator.php

```diff
@@ -0,0 +1,92 @@
+<?php
+namespace ListingsApp\Classes;
+use PDO;
+class DBPopulator {
+    private ListingAPI $listingAPI;
+    private PDO $dbConnection;
+    private array $listings;
+    /**
+     * DBPopulator constructor.
+     * @param $listingAPI
+     * @param $db
+     */
+    public function __construct(ListingAPI $listingAPI, PDO $db) {
+        $this->listingAPI = $listingAPI;
+        $this->dbConnection = $db;
+        $this->populateDBAllTables();
+    }
+
+    /**
+     * fills listings table in listings-feed db
+     * @return void
+     */
+    private function populateDBListingsTable(): void {
```

Comment:
Suggest switching all of these methods to be protected instead of public - that way in future this class could be extended and tweaked without having to update the original.

This is currently violating the open-closed principle of SOLID